test/workflow_test.go

```diff
@@ -39,7 +39,9 @@ const (
 	consistentQuerySignalCh = "consistent-query-signal-chan"
 )
 
-type Workflows struct{}
+type Workflows struct {
+	nonDeterminismSimulatorWorkflowCallCount int
```

Comment:
since this struct is kinda intended for multi-test use... I wonder if this'll come back to bite us some day.  as compared to making a different struct each time to ensure they're isolated.

but eh.  the toggling is a pretty simple way to achieve this even in that case, seems probably fine.

Reply:
I think isolated test cases are going to be less error prone so when we refactor we would actually do everything inside `SetupSuite` per test case.